homework/index.js

```diff
@@ -29,26 +37,163 @@
     return elem;
   }
 
+  function appendRepoDetail(header, description, parentTable, link) {
+    const repoDetailRow = createAndAppend('tr', parentTable);
+    createAndAppend('th', repoDetailRow, { text: header });
+    const descriptionElm = createAndAppend('td', repoDetailRow);
+    if (link) {
+      createAndAppend('a', descriptionElm, {
+        href: link,
+        text: description,
+        target: '_blank',
+      });
+    } else {
+      descriptionElm.textContent = description;
+    }
+    return repoDetailRow;
+  }
+
+  function getDateTimeText(dateTimeStr) {
+    const dateTime = new Date(dateTimeStr);
+    return dateTime.toLocaleString();
+  }
+
+  function sortRepositoriesByNameAscending(firstRepo, secondRepo) {
+    return firstRepo.name.localeCompare(secondRepo.name);
+  }
+
+  function renderRepositoryNames(repositories, select) {
+    repositories.forEach((repository, index) => {
+      createAndAppend('option', select, {
+        text: repository.name,
+        value: index,
+      });
+    });
+  }
+
+  function sortAndMapRepositories(repositoryInfos) {
+    return repositoryInfos
+      .sort(sortRepositoriesByNameAscending)
+      .map(repositoryInfo => ({
+        name: repositoryInfo.name,
+        url: repositoryInfo.html_url,
+        description: repositoryInfo.description,
+        forks: repositoryInfo.forks,
+        lastUpdateTime: getDateTimeText(repositoryInfo.updated_at),
+        contributorsUrl: repositoryInfo.contributors_url,
+      }));
+  }
+
   function renderRepoDetails(repo, ul) {
-    createAndAppend('li', ul, { text: repo.name });
+    const repoDetailsListItem = createAndAppend('li', ul, {
```

Comment:
You are now rendering the details for a single repository, so it no longer makes sense to use an unordered list(with a single list item). You can make your table a direct child of the parent section.